ocaml/xapi/xapi_xenops.ml

```diff
@@ -2281,13 +2284,25 @@ let update_vm ~__context id =
               try
                 Option.iter
                   (fun (_, state) ->
-                    debug "xenopsd event: Updating VM %s last_start_time <- %s"
-                      id
-                      (Date.to_string (Date.of_float state.Vm.last_start_time)) ;
                     let metrics = Db.VM.get_metrics ~__context ~self in
                     let start_time = Date.of_float state.Vm.last_start_time in
-                    Db.VM_metrics.set_start_time ~__context ~self:metrics
-                      ~value:start_time ;
+                    if
+                      start_time
+                      <> Db.VM_metrics.get_start_time ~__context ~self:metrics
+                    then (
+                      debug
+                        "xenopsd event: Updating VM %s last_start_time <- %s" id
+                        (Date.to_string (Date.of_float state.Vm.last_start_time)) ;
+                      Db.VM_metrics.set_start_time ~__context ~self:metrics
+                        ~value:start_time ;
+                      if
+                        power_state <> `Suspended (* not VM suspend *)
+                        && power_state_before_update
+                           <> `Suspended (* not VM resume *)
+                      then
+                        Xapi_vm_lifecycle.remove_pending_guidance ~__context
```

Comment:
Would it always be safe to clear the `restart_device_model` here?
(Otherwise, `RestartVM` might not supersede the `RestartDeviceModel`)

Reply:
Yes, I think it should be safe to also clear `restart_device_model` here to make sure `restart_device_model` would always be when VM is restarted.
But if we indeed have a scenario which misses changing `domid` on VM restart, there should be a bug need to be fixed.

Reply:
`But if we indeed have a scenario which misses changing domid on VM restart`
I don't think this will be a valid case. And I think your tests also confirmed this. It's agreed that RestartVM supersedes RestartDeviceModel.

Reply:
I mean there should not have such case, as once a VM is restarted, its `domid` will always be changed.